parser/src/command/glacier.rs

```diff
@@ -0,0 +1,116 @@
+//! The glacier command parser.
+//!
+//! This adds the option to track ICEs.
+//!
+//! The grammar is as follows:
+//!
+//! ```text
+//! Command: `@bot glacier add <code-source>?`
+//!
+//! <code-source>:
+//!   - "https://play.rust-lang.org/.*"
+//! ```
+
+use std::fmt;
+use crate::error::Error;
+use crate::token::{Token, Tokenizer};
+use regex::Regex;
+
+#[derive(PartialEq, Eq, Debug)]
+pub struct GlacierCommand {
+    source: String
+}
+
+#[derive(PartialEq, Eq, Debug)]
+pub enum ParseError {
+    NoLink,
+    InvalidLink,
+}
+
+impl std::error::Error for ParseError {}
+
+impl fmt::Display for ParseError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::NoLink => write!(f, "no link provided"),
+            Self::InvalidLink => write!(f, "invalid link"),
+        }
+    }
+}
+
+impl GlacierCommand {
+    pub fn parse<'a>(input: &mut Tokenizer<'a>) -> Result<Option<GlacierCommand>, Error<'a>> {
+        let mut toks = input.clone();
+        if let Some(Token::Word("glacier")) = toks.peek_token()? {
+            toks.next_token()?;
+            match toks.next_token()? {
+                Some(Token::Quote(s)) => {
+                    let source = s.to_owned();
+                    let playground_url = Regex::new("https://play.rust-lang.org/.*").unwrap();
```

Comment:
Why not using [`starts_with`](https://doc.rust-lang.org/std/primitive.str.html#method.starts_with)? I don't like using a regex for this case, because you pull up the crate just this piece and you could circumvent it.

Reply:
That's a very good suggestion! It didn't cross my mind, heh.